lib/api_endpoint/swagger2/api_swagger_parameter_handler.py

```diff
@@ -83,7 +83,16 @@ def wrap_body_params(
                 enum_svc,
                 ref_path)
             parameter_obj['description'] = param.documentation
-            body_obj.update(parameter_obj)
+            if 'BodyField' in param.metadata:
+                body_obj['type'] = 'object'
+                body_obj['properties'] = properties_obj
+                properties_obj[param.metadata['BodyField'].elements['name'].string_value] = parameter_obj
+                if 'required' not in parameter_obj:
+                    required.append(param.name)
```

Comment:
Are you sure that if required is not in parameter_obj, then the param name has to be added to the required list?

Sounds opposite of common sense.

Reply:
The required comes as `required=false` when it is not required in the BodyField case.